problem_builder/mentoring.py

```diff
@@ -355,12 +355,27 @@ def allowed_nested_blocks(self):
         except ImportError:
             pass
 
+        message_block_shims = [
+            NestedXBlockSpec(
+                MentoringMessageBlock,
+                category='pb-message',
+                boilerplate=message_type,
+                label=get_message_label(message_type),
```

Comment:
The Message block specs should have `single_instance=True`, because we only allow a single instance of each message type inside a Mentoring block, *BUT* I tried that and it looks like `single_instance` functionality is currently [implemented in a rather hackish way](https://github.com/edx/xblock-utils/blob/7e7e6fc20e2de3ecb0c882db2bde45b5e4d1ccbb/xblockutils/public/studio_container.js#L40-L48) and it can only limit blocks by category. Here we still want to allow multiple `pb-message` blocks, as long as they are of different `message_type`, but it doesn't look like we can do that currently.

If you set `single_instance=True`, the button that you click gets correctly disabled when adding a Message type block to a mentoring container, but if you refresh the page, *all* `pb-message` buttons are disabled rather than just the one type that you added to the container.

One way to fix this would be to introduce subclasses of `MentoringMessageBlock`, so that each message type would be a separate class.
Another way would be to expand the `single_instance` feature of nested xblocks in xblock-utils so that it would be able to toggle the button on parameters other than just the block category.

I think I prefer the second option, but neither of these options are simple, so in order to not block this PR, I think I am fine with leaving `single_instance=True` out for now, and create a task to handle this in next sprint.